save-backend/src/main/kotlin/com/saveourtool/save/backend/service/OrganizationService.kt

```diff
@@ -62,13 +67,19 @@ class OrganizationService(
      * Mark organization [organization] as deleted
      *
      * @param organization an [Organization] to delete
+     * @param authentication
      * @return deleted organization
      */
-    fun deleteOrganization(organization: Organization): Organization = if (!hasProjects(organization.name)) {
-        changeOrganizationStatus(organization, OrganizationStatus.DELETED)
-    } else {
-        organization
-    }
+    fun deleteOrganization(organization: Organization, authentication: Authentication): Organization =
+            if (!hasProjects(organization.name) || organizationPermissionEvaluator.hasPermission(authentication, organization, Permission.BAN)) {
+                projectService.getAllByOrganizationName(organization.name).forEach {
+                    it.status = DELETED
+                    projectService.updateProject(it)
+                }
+                changeOrganizationStatus(organization, OrganizationStatus.DELETED)
+            } else {
+                organization
```

Comment:
you return the same organization if user doesn't have permission to delete\ban this organization.
I think we should throw exception in such cases

Reply:
Deleted this logic